./configure --disable-libcurl fails

Hi,

I'm trying to install samtools and bcftools version 1.22 on a server.

When running `./configure` I get the following error message:

```
configure: WARNING: libcurl not enabled: library not found
configure: WARNING: GCS support not enabled: requires libcurl support
configure: WARNING: S3 support not enabled: requires libcurl support
configure: error: ref-cache not enabled

The ref-cache program requires libcurl support to be enabled in HTSlib.
Configure with --enable-libcurl in order to build ref-cache.
configure: error: ./configure failed for htslib-1.22
```

When running `./configure --enable-libcurl`:

```
configure: error: libcurl library not found

Support for HTTPS and other SSL-based URLs requires routines from the libcurl
library <http://curl.se/libcurl/>.  Building HTSlib with libcurl enabled
requires libcurl development files to be installed on the build machine; you
may need to ensure a package such as libcurl4-{gnutls,nss,openssl}-dev (on
Debian or Ubuntu Linux) or libcurl-devel (on RPM-based Linux distributions
or Cygwin) is installed.

Either configure with --disable-libcurl or resolve this error to build HTSlib.
configure: error: ./configure failed for htslib-1.22

```

Then, if `./configure --disable-libcurl`, first error shows again:

```
configure: WARNING: GCS support not enabled: requires libcurl support
configure: WARNING: S3 support not enabled: requires libcurl support
configure: error: ref-cache not enabled

The ref-cache program requires libcurl support to be enabled in HTSlib.
Configure with --enable-libcurl in order to build ref-cache.
configure: error: ./configure failed for htslib-1.22

```

I don't have admin privileges on this server, nor do I need libcurl capabilities, so I wouldn't want to have to install and configure a static version of libcurl.

Version 1.21 installs just fine on the same machine, even if the warnings still appear:

```
configure: WARNING: libcurl not enabled: library not found
configure: WARNING: GCS support not enabled: requires libcurl support
configure: WARNING: S3 support not enabled: requires libcurl support

```

Thanks a lot.

--- a/configure.ac
+++ b/configure.ac
@@ -645,7 +645,7 @@ AS_IF([test "x$enable_ref_cache" != xno],
   [AS_CASE([$PLATFORM],
      [Darwin | default],[
         AS_IF([test "x$libcurl" = xenabled], [ref_cache="enabled"], [
-          AS_IF([test "x$enable_ref_cache" = check], [
+          AS_IF([test "x$enable_ref_cache" = xcheck], [
             AC_MSG_WARN([ref-cache not enabled: requires libcurl])
           ],[
             MSG_ERROR([ref-cache not enabled

